### PR TITLE
Remove prefix from script tag helper attribute "src"

### LINKED ISSUE #28818: Fix asp-src
[asp-src](https://learn.microsoft.com/en-us/aspnet/core/mvc/views/tag-helpers/built-in/script-tag-helper?branch=pr-en-us-28802&view=aspnetcore-7.0#asp-src)

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: c90104e1-65b1-0007-e56d-0dffef4b1a44
* Version Independent ID: 5c95e5dc-dc10-8658-91fd-1d55d3959c37
* Content: [Script Tag Helper in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/mvc/views/tag-helpers/built-in/script-tag-helper?branch=pr-en-us-28802&view=aspnetcore-7.0#asp-src)
* Content Source: [aspnetcore/mvc/views/tag-helpers/built-in/script-tag-helper.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/mvc/views/tag-helpers/built-in/script-tag-helper.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by cn-byn**
Document the script tag helper attribute "src" instead of the non-existent attribute "asp-src". Move description of said attribute to the top, similar to link tag helper's "href" attribute.

Fixes #28818

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/mvc/views/tag-helpers/built-in/script-tag-helper.md](https://github.com/dotnet/AspNetCore.Docs/blob/9c97679221b65da183bd31bce0b2984d77638432/aspnetcore/mvc/views/tag-helpers/built-in/script-tag-helper.md) | [aspnetcore/mvc/views/tag-helpers/built-in/script-tag-helper](https://review.learn.microsoft.com/en-us/aspnet/core/mvc/views/tag-helpers/built-in/script-tag-helper?branch=pr-en-us-28802) |

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img
that's HTML, not a tag helper.

**Comment by cn-byn** [comment]
> See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img that's HTML, not a tag helper.

@Rick-Anderson Excuse me? The ASP.NET Core documentation for the class `ScriptTagHelper` mentions an attribute named "asp-src", which does not exist.

It does, however, NOT mention the attribute "src", [which does exist](https://github.com/dotnet/aspnetcore/blob/main/src/Mvc/Mvc.TagHelpers/src/ScriptTagHelper.cs) and is the one to be used:

```cs
// ScriptTagHelper.cs:
//
/// <summary>
/// Address of the external script to use.
/// </summary>
/// <remarks>
/// Passed through to the generated HTML in all cases.
/// </remarks>
[HtmlAttributeName(SrcAttributeName)]
public string Src { get; set; }
```

Of course that's an HTML attribute. It's just the documentation that is wrong. "asp-src" cannot be used.

The documentation for `LinkTagHelper` mentions its related attribute "href" in the docs, which obviously is also an HTML attribute.

image

```cs
// LinkTagHelper.cs:
//
/// <summary>
/// Address of the linked resource.
/// </summary>
/// <remarks>
/// Passed through to the generated HTML in all cases.
/// </remarks>
[HtmlAttributeName(HrefAttributeName)]
public string Href { get; set; }
```

**Comment by Rick-Anderson** [comment]
[public string Src { get; set; }](https://github.com/dotnet/aspnetcore/blob/main/src/Mvc/Mvc.TagHelpers/src/ScriptTagHelper.cs#L116)

File: aspnetcore/mvc/views/tag-helpers/built-in/script-tag-helper.md [hunk 1]
```diff
@@ -34,6 +34,10 @@ Don't use the `<script>` element's [defer](https://developer.mozilla.org/docs/We
 
 See [Script Tag Helper](xref:Microsoft.AspNetCore.Mvc.TagHelpers.ScriptTagHelper) for all the Script Tag Helper attributes, properties, and methods.
 
+### src
+
+Address of the external script to use.
+
 ### asp-append-version
 
 When `asp-append-version` is specified with a `true` value along with a `src` attribute, a unique version is generated.
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/views/tag-helpers/built-in/script-tag-helper.md:43]:
```suggestion
When `asp-append-version` is specified with a `true` value along with a [`src`](https://github.com/dotnet/aspnetcore/blob/main/src/Mvc/Mvc.TagHelpers/src/ScriptTagHelper.cs#L116) attribute, a unique version is generated.
```

**Comment by Rick-Anderson** [comment]
@andrewlock @pkellner please review

**Comment by pkellner** [comment]
Looks good to me though I don't think I wrote that particular doc on the script tag helper. I can't find any other discussion of asp-src so sounds like a mistake. Maybe got confused with asp-fallback-src?

**Comment by Rick-Anderson** [comment]
@cn-byn  
Thanks for the contribution. Be sure to check out your name as a contributor about the 3rd of next month on the [What's new page](https://docs.microsoft.com/en-us/aspnet/core/whats-new/) in the **Community contributors** section at the bottom.

Check out [is:issue is:open label:"code help wanted" "](https://github.com/dotnet/AspNetCore.Docs/issues?q=is%3Aissue+is%3Aopen+label%3A%22code+help+wanted%22+) and [is:open is:issue label:"Good first issue"](https://github.com/dotnet/AspNetCore.Docs/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+first+issue%22)

**Pull request merged**